src/main/java/gg/solarmc/clans/command/commands/clans/InfoCommand.java

```diff
@@ -74,21 +74,28 @@ private void sendClanInfo(CommandSender player, Clan clan, PluginHelper helper)
         int deaths = clan.currentDeaths();
         StringBuilder members = new StringBuilder();
 
-        CentralisedFuture[] memberFutures = clan.currentMembers().stream().map(it ->
-                getPlayerNameById(player.getServer(), it.userId())
-                        .thenAccept(a -> members.append(a).append(",")))
-                .toArray(CentralisedFuture[]::new);
-
-        helper.futuresFactory(player.getServer()).allOf(memberFutures).thenRunSync(() -> {
-            Component info = helper.replaceText(plugin.getPluginConfig().clan().info(),
-                    Map.of("{clan}", clanName,
-                            "{kills}", String.valueOf(kills),
-                            "{assists}", String.valueOf(assists),
-                            "{deaths}", String.valueOf(deaths),
-                            "{ally}", allyClanName,
-                            "{members}", members.toString()));
-
-            player.sendMessage(info);
+        final Server server = player.getServer();
+
+        server.getDataCenter().runTransact(transaction -> {
+            CentralisedFuture[] memberFutures = clan.getClanMembers(transaction).stream().map(it ->
+                    getPlayerNameById(server, it.userId())
+                            .thenAccept(a -> members.append(a).append(",")))
```

Comment:
StringBuilder is not thread-safe, use StringBuffer for thread-safety

Reply:
replacing StringBuilder with StringBuffer won't be sufficient, actually. You'll need to combine the two calls to append or synchronize when using them both.

Reply:
combine the two calls to append or synchronize when using them both???? combine which 2 calls ?

Reply:
If you can put your builder in the context of a single transaction and do not let it escape rather than the unsafe supercontext of this method, you should have something more threadsafe.